### ISSUE TITLE
Debugger Error CS7069: Reference to type 'DbConnectionWrapper' claims it is defined in 'Insight.Database', but it could not be found

### ISSUE THREAD

**Issue by shaxxx** [jonwagner/Insight.Database#224]
When opening connection with transaction debbuger is unable to view any variable values.
Watch returns Error CS7069.
Tested on Windows 10, VS2015 Community Edition with lastest NuGet packages.
Also tried pulling the source from current branch but issue persist.

Here's simple example
```
var database = new SqlConnectionStringBuilder(connectionString);
using (var tx = database.OpenWithTransaction())
    {
        //Here we can see variable values just fine, tx has properties, connection is opened
        var myList = tx.QuerySql<MyType>("SELECT * FROM Table"); // <-- Query is sucessfull, but Debugger goes "blind"
       / /Query returns list just fine but Watch returns error CS7069
        tx.Commit();
    }
```

Seems it goes "blind" after calling 
StaticFieldStorage.EmitLoad(il, provider);
in DbParameterGenerator.cs on line 208.

Anything that can be done?

**Comment by janus007** [comment]
Reminds me about:
error CS7069: Reference to type 'FastExpando' claims it is defined in 'Insight.Database', but it could not be found

I've strong feeling that Jon will fix it asap, in the meantime I will use Insight for inserts and if needed I will resort to the old days :)

**Comment by msieker** [comment]
As another data point, version 5.2.7.0 from NuGet exhibits the same behavior on VS2015 Enterprise. Before this statement, the debugger works:
```C#
_conn.Query("CreateSubmission",
                new { MaxRecords = maxRecords },
                Query.Returns(Some<Submission>.Records)
                    .Then(Some<BatchDetail>.Records)
                    .Then(Some<AccessionTest>.Records)
                    .ThenChildren(Some<AccessionTestResult>.Records),
                transaction: _tran)
```
Once this returns, the debugger displays `error CS7069: Reference to type 'Results<,,>' claims it is defined in 'Insight.Database', but it could not be found` for all variables in the function, even ones unrelated to the results data set.

Things that have been tried:
 - Enabling or disabling Intellitrace
 - Enabling or disabling Just My Code

**Comment by GeorgeKnap** [comment]
Hi. Getting the same issue.
Insight.Database v 5.2.7 ; Visual Studio Professional 2015.
When using method `QueryResultAsync<T1,...>(...)` my VS debugger gets confused and throw this when trying to watch the variables values.

`error CS7069: Reference to type 'Results<,,,,>' claims it is defined in 'Insight.Database', but it could not be found.`

No issues when using synchronous `Query<T>(...)`

**Comment by shaxxx** [comment]
Any news about bug fixes or workarounds? This is a big deal for us.

**Comment by rglaspey** [comment]
I am hitting this problem too. Need a way to resolve, please.

**Comment by shaxxx** [comment]
I was able to pinpoint the problem. In the StaticFieldStorage.cs file, when constructor is called, on line
`AssemblyBuilder ab = AppDomain.CurrentDomain.DefineDynamicAssembly(an, AssemblyBuilderAccess.Run);`

Now, the problem with the debbugger occurs only if BOTH conditions are satisfied

1.  AssemblyName parameter (an) has it's name property same as the Assembly.GetExecutingAssembly().GetName().Name;
2. AssemblyBuilderAccess parameter is set to Run or RunCollect

If either of this is false debugger will work as expected.
Quick and dirty (but working) workaround is to simply use AssemblyName.Name that's different than the one from Assembly.GetExecutingAssembly().GetName().Name.
In other words, to get it working in file  StaticFieldStorage.cs after line 37 insert something like this
`an.Name += ".Tmp";`
I've tested this with async inserts, updates, deletes, batch inserts on MSSQL server and everything works. I assume if you're using auto implemented interfaces you'll need to change constructor in InterfaceGenerator.cs file as well.

Now, this currently work for us, but I would be sooo happy if @jonwagner could take a look at this and give us his opinion.

Be advised, this debugger quirk is not result of Insight database code. I've tested these 3 lines in a separate project and debugger still goes dark afterwards. Still, would be nice to have this work out of the box.

**Comment by AndrewFreese** [comment]
Not sure how helpful this is, but if I switch back to VS2013, i can debug normally.

**Comment by keerthiramk** [comment]
Yes, I am having the same issue when I query data from the database through 'QueryResults' method using VS 2015. 
But it works normally with VS 2013.

**Comment by fomenta** [comment]
Fixed bug on Visual Studio 2015.
Go to Tools menu, Options, Debugging, General: Check: "Use the legacy C# and VB expression evaluators"

**Comment by tshepherdcirrus** [comment]
Can confirm the Visual Studio setting fix posted by fomenta worked for me

**Comment by AndrewFreese** [comment]
Though the visual studio setting is a big win, the TINY code change shaxxx proposed would be even better. Any change this can be reopened?

**Comment by shaxxx** [comment]
Now when Jon and the new crew is back, I totally agree. 
Please update the project so it works out of the box in VS2015.

**Comment by kevinsea** [comment]
Nice catch shaxxx.  I agree, this should be fixed.  

Do you want to submit the pull request?  If not I'd happy to submit one.

**Comment by shaxxx** [comment]
Done.

**Comment by jonwagner** [comment]
Hm...

I suspect that the issue is that when we load a new assembly with the old name, the debugger is overwriting its symbol table.

If we change the name of the dynamic assembly, then we'll break anyone that's using `InternalsVisibleTo(Insight.Database)` I don't mind a breaking change if we do it in the "6.0" release that will also have .NET core support and some release notes.

We'd want a better name than `Insight.Database.tmp`. Maybe `InternalsVisibleTo(Insight.Database.DynamicAssembly`?

I don't know if there's a good test case for this, so we can just do a PR for the code suggested with a good name and PR it into the 6.0 branch.

**Comment by kevinsea** [comment]
@jonwagner, the discussion continues in @267 with the same angst about 6.0 and a potential solution.

**Comment by kevinsea** [comment]
Hey @shaxxx, your fix has been released in 5.2.8-Beta, please check it out.

**Comment by jonwagner** [comment]
Fixed in v5.2.8
